statsmodels/tsa/tests/test_tsa_indexes.py

```diff
@@ -307,6 +322,20 @@ def test_instantiation_valid():
         assert_equal(mod.data.dates, None)
         assert_equal(mod.data.freq, None)
 
+        if has_range_index:
```

Comment:
Can this become a separate test?  flake8 complains about this test being too complicated already

Reply:
```
@pytest.mark.skipif(not has_range_index, reason="RangeIndex not available with older pandas")
def test_instantiation_valid_range_index():
     for base_endog in dta[2:4]:
```

... followed by lines 326-337, but with one less level of indentation.

Also if you're interested in a copy/paste option to split up this giant test (pre-this-PR), [here ya go](https://github.com/jbrockmendel/sm2/blob/master/sm2/tsa/tests/test_tsa_indexes.py.